File: ruby/testcode/redirect_037.rb
require_relative 'shared'

# vuln-code-snippet start ruby_redirect_login_fixed
def redirect_login_fixed(req)
  BenchmarkResponse.new(302, { 'Location' => '/dashboard' }, '') # vuln-code-snippet safe-line ruby_redirect_login_fixed
end
# vuln-code-snippet end ruby_redirect_login_fixed

File: php/testcode/redirect_037.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_redirect_numeric_id_map
function redirect037(BenchmarkRequest $req): BenchmarkResponse {
    $id = intval($req->param('id'));
    $urls = [1 => '/home', 2 => '/profile', 3 => '/settings'];
    $url = $urls[$id] ?? '/'; // vuln-code-snippet safe-line php_redirect_numeric_id_map
    header('Location: ' . $url);
    return BenchmarkResponse::ok('');
}
// vuln-code-snippet end php_redirect_numeric_id_map
